src/main/scala/org/ergoplatform/ErgoAddress.scala

```diff
@@ -203,6 +205,17 @@ case class ErgoAddressEncoder(networkPrefix: Byte) {
       }
     }
   }
+
+  def fromPropositionBytes(bytes: Array[Byte]): Try[ErgoAddress] = Try {
```

Comment:
@oskin1 Could you provide some context, where do you want to use this operation? It looks like an address deserializer to me. We have a special approach to (de)serializer implementation (see [ProveDlog serializer](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/ae50313900d65fcfaaf58e35efd9c9f93c07fda5/src/main/scala/sigmastate/serialization/ProveDlogSerializer.scala#L11))
Also guessing the address type seems fragile. We have address type prefixes in the `toString/fromString`.

Reply:
@greenhat, this method will be used in explorer, where we have only serialised script bytes - we need to match these bytes by leading opcodes or some mask to distinct different address types and convert particular script to suitable ErgoAddress type.